editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/wizards/pages/GlobalBeanEditWizardPage.java

```diff
@@ -0,0 +1,231 @@
+		UpdateValueStrategy strategy = new UpdateValueStrategy();
+		strategy.setBeforeSetValidator(new NewBeanIdPropertyValidator(CamelBean.PROP_ID, parent));
+
+		ComputedValue<?> idValue = new ComputedValue<Object>() {
+
+			@Override
+			protected Object calculate() {
+				return beanConfigUtil.getAttributeValue(selectedElement, CamelBean.PROP_ID);
+			}
+
+			@Override
+			protected void doSetValue(Object value) {
+				final String strValue = (String) value;
+				final String oldValue = (String) beanConfigUtil.getAttributeValue(selectedElement, CamelBean.PROP_ID);
+				if (!oldValue.contentEquals(strValue)) {
+					beanConfigUtil.setAttributeValue(selectedElement, CamelBean.PROP_ID, strValue);
+				}
+				getValue();
+			}
+		};
+		Binding binding = dbc.bindValue(WidgetProperties.text(SWT.Modify).observe(idText), idValue, strategy, null);
+		ControlDecorationSupport.create(binding, SWT.LEFT | SWT.TOP);
+	}
+
+	/**
+	 * @param composite
+	 */
+	private void createClassLine(Composite composite) {
+		Label classLabel = new Label(composite, SWT.NONE);
+		classLabel.setText("Class");
+		Text classText = new Text(composite, SWT.BORDER);
+		classText.setLayoutData(GridDataFactory.fillDefaults().indent(10, 0).grab(true, false).create());
+		UpdateValueStrategy strategy = new UpdateValueStrategy();
+		strategy.setBeforeSetValidator(new BeanClassExistsValidator());
+
+		ComputedValue<?> classValue = new ComputedValue<Object>() {
+
+			@Override
+			protected Object calculate() {
+				return beanConfigUtil.getAttributeValue(selectedElement, CamelBean.PROP_CLASS);
+			}
+
+			@Override
+			protected void doSetValue(Object value) {
+				final String strValue = (String) value;
+				final String oldValue = (String) beanConfigUtil.getAttributeValue(selectedElement, CamelBean.PROP_ID);
+				if (!oldValue.contentEquals(strValue)) {
+					beanConfigUtil.setAttributeValue(selectedElement, CamelBean.PROP_CLASS, strValue);
+				}
+				getValue();
+			}
+		};
+		Binding binding = dbc.bindValue(WidgetProperties.text(SWT.Modify).observe(classText), classValue, strategy, null);
+		ControlDecorationSupport.create(binding, SWT.LEFT | SWT.TOP);
+	}
+
+	private void createBrowseButton(Composite composite) {
+		Button browseBeanButton = new Button(composite, SWT.PUSH);
+		browseBeanButton.setText("...");
+		browseBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				final IProject project = CamelUtils.project();
+				String value = beanConfigUtil.handleClassBrowse(project, getShell());
+				if (value != null) {
+					classObservable.setValue(value);
+				}
+			}
+		});
+	}
+
+	private void createClassNewButton(Composite composite) {
+		Button newBeanButton = new Button(composite, SWT.PUSH);
+		newBeanButton.setText("+");
+		newBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				final IProject project = CamelUtils.project();
```

Comment:
better to avoid to use the CamelUtils.project() as it is linking this code the workbench state.
On AbstractCamelElement we can getCamelFile().getResource().getProject()